pyramid_zipkin/thrift_helper.py

```diff
@@ -153,14 +153,15 @@ def thrift_obj_in_bytes(thrift_obj):  # pragma: no cover
     return bytes(trans.getvalue())
 
 
-def get_id(id_value):
-    """
-    Get the integer version of an encoded zipkin ID
+def unsigned_hex_to_signed_int(hex_string):
+    """Converts a 64-bit hex string to a signed int value.
 
-    :param id_value: the string representation of a zipkin ID
-    :returns: int representation of zipkin ID
-    """
-    if len(id_value) == 0:
-        return 0
+    This is due to the fact that Apache Thrift only has signed values.
 
-    return int(id_value, 16)
+    :param hex_string: the string representation of a zipkin ID
+    :returns: signed int representation
+    """
+    # Backwards compatibility with ids generated by previous versions
+    if hex_string.startswith('-0x') or hex_string.startswith('0x'):
+        return int(hex_string, 16)
+    return struct.unpack('q', struct.pack('Q', int(hex_string, 16)))[0]
```

Comment:
Why can't you do `int(hex_string, 16)` in both cases? It looks to me like that works. I.e. `int('0xff', 16) == int('ff', 16)`, in both 2.7 and 3.4, AFAICT.